### PR TITLE
[WIP] add epsilon greedy and boltzmann exploration

### PR DESCRIPTION
**Pull request by ajaysub110**

### CODE REVIEW

**Comment by ajaysub110** [comment]
I'm thinking of using decorators for this. The decorator for an exploration method will take in the current action selection method and modify it by adding noise/exploration. Your thoughts @Sharad24 ?

File: genrl/deep/common/exploration.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# create decorator functions for exploration methods
+# decorator takes as param the select_action method and adds noise to it
+# Some exploration methods require networks/access to the model
+import numpy as np
+
+import gym
+
+class EpsilonGreedyDecorator:
```
Comment (Sharad24) [inline genrl/deep/common/exploration.py:8]: As a user, I would want the decorator name to be short, so can you make this shorter?

Comment (Sharad24) [reply genrl/deep/common/exploration.py:8]: I think it can just be `epsgreedy`, we dont need `decorator` in the name.

**Comment by Sharad24** [comment]
Still working on this?

**Comment by ajaysub110** [comment]
Yeah, will take some more time cause I've been pretty busy recently

**Comment by ajaysub110** [comment]
I guess someone else can take this at this point..

**Comment by Sharad24** [comment]
😭

**Comment by ajaysub110** [comment]
😢

**Pull request closed without merging**